docs/running-locust-distributed.rst

```diff
@@ -87,6 +87,44 @@ listen to. Defaults to 5557.
 Used when starting the master node with ``--headless``. The master node will then wait until X worker
 nodes has connected before the test is started.
 
+Communicating across nodes
+=============================================
+
+When running Locust in distributed mode, you may want to communicate between nodes in order to coordinate 
+data. This can be easily accomplished with custom messages using the built in messaging hooks:
+
+.. code-block:: python
+
+    from locust import events
+    from locust.runners import MasterRunner, WorkerRunner
+
+    # Fired when the worker recieves a message of type 'test_users'
+    def setup_test_users(environment, msg, **kwargs):
+        for user in msg.data:
+            print(f"User {user['name']} recieved")
+        environment.runner.send_message('acknowledge_users', f"Thanks for the {len(msg.data)} users!")
+
+    # Fired when the master recieves a message of type 'acknowledge_users'
+    def on_acknowledge(msg, **kwargs):
+        print(msg.data)
+
+    @events.init.add_listener
+    def on_locust_init(environment, **_kwargs):
+        if isinstance(environment.runner, WorkerRunner):
+            environment.runner.register_message('test_users', setup_test_users)
+        elif isinstance(environment.runner, MasterRunner):
+            environment.runner.register_message('acknowledge_users', on_acknowledge)
+
+    @events.test_start.add_listener
+    def on_test_start(environment, **_kwargs):
+        if isinstance(environment.runner, MasterRunner):
+            users = [
+                {"name": "User1"},
+                {"name": "User2"},
+                {"name": "User3"},
+            ]
+            environment.runner.send_message('test_users', users)  
+
```

Comment:
Maybe add an actual User class, so that the example can run for real?

Reply:
Most inline code blocks in the documentation seem to only include the relevant code snippets, I would think to avoid cluttering up the page with familiar concepts. What about adding just a link to the full example in the examples directory?

Reply:
Sounds good!